### DISCUSSION TITLE
how to remove backdrop-filter for Firefox?

### DISCUSSION THREAD

**Discussion by nekooee** [tailwindlabs/tailwindcss#10987] [Help] [answered]
I noticed today that my site is messed up in Firefox and some parts are not showing. After checking, I realized that the problem is "backdrop-blur", which is not yet supported in Firefox.
But how to define that it will be displayed only in the supported browsers and not for Firefox?

**Comment by wongjn** [comment]
You can use [`supports-[]`](https://tailwindcss.com/docs/hover-focus-and-other-states#supports-rules) to use feature detection with [`@supports`](https://developer.mozilla.org/en-US/docs/Web/CSS/@supports) to detect if a CSS property is supported. For example, you could do `supports-[backdrop-filter]:backdrop-blur`.

As an aside, [MDN reports that Firefox 103+ should support `backdrop-filter`](https://developer.mozilla.org/en-US/docs/Web/CSS/backdrop-filter#browser_compatibility).

**Reply by wongjn** [reply]
> Thankful. I wrote it in the way you said, but it is still applied in Firefox. my class: `supports-[backdrop-filter]:tw-backdrop-blur-sm`
> 
> and result in output css:
> 
> ```
> @supports ((-webkit-backdrop-filter: var(--tw)) or (backdrop-filter: var(--tw))) {
>   .supports-\[backdrop-filter\]\:tw-backdrop-blur-sm {
>     --tw-backdrop-blur: blur(4px);
>     -webkit-backdrop-filter: var(--tw-backdrop-blur) var(--tw-backdrop-brightness) var(--tw-backdrop-contrast) var(--tw-backdrop-grayscale) var(--tw-backdrop-hue-rotate) var(--tw-backdrop-invert) var(--tw-backdrop-opacity) var(--tw-backdrop-saturate) var(--tw-backdrop-sepia);
>             backdrop-filter: var(--tw-backdrop-blur) var(--tw-backdrop-brightness) var(--tw-backdrop-contrast) var(--tw-backdrop-grayscale) var(--tw-backdrop-hue-rotate) var(--tw-backdrop-invert) var(--tw-backdrop-opacity) var(--tw-backdrop-saturate) var(--tw-backdrop-sepia);
>   }
> }
> ```
> 
> and in firefox:
> 
> ![image](https://user-images.githubusercontent.com/6003803/231226484-e95d4515-ec9d-44fe-97b5-887ceaa923c3.png)[^1]

It still applies because as stated previously, and as can be seen in your developer tools screenshot, Firefox *does* support `backdrop-filter`.

[^1]: https://github.com/tailwindlabs/tailwindcss/discussions/10987#discussioncomment-5583066

**Reply by nekooee** [reply]
My firefox is version 112, but it still does not support it.

**Reply by nekooee** [reply]
According to a user in the link below, the support option no longer works in Firefox version 111+ and returns true, while it still does not work properly.
https://gist.github.com/mrtcmn/0490fb79fa63418e4409a99553f80e26?permalink_comment_id=4504075#gistcomment-4504075

**Comment by nekooee** [comment]
Thankful. I wrote it in the way you said, but it is still applied in Firefox. 
my class: `supports-[backdrop-filter]:tw-backdrop-blur-sm`

and result in output css:
```
@supports ((-webkit-backdrop-filter: var(--tw)) or (backdrop-filter: var(--tw))) {
  .supports-\[backdrop-filter\]\:tw-backdrop-blur-sm {
    --tw-backdrop-blur: blur(4px);
    -webkit-backdrop-filter: var(--tw-backdrop-blur) var(--tw-backdrop-brightness) var(--tw-backdrop-contrast) var(--tw-backdrop-grayscale) var(--tw-backdrop-hue-rotate) var(--tw-backdrop-invert) var(--tw-backdrop-opacity) var(--tw-backdrop-saturate) var(--tw-backdrop-sepia);
            backdrop-filter: var(--tw-backdrop-blur) var(--tw-backdrop-brightness) var(--tw-backdrop-contrast) var(--tw-backdrop-grayscale) var(--tw-backdrop-hue-rotate) var(--tw-backdrop-invert) var(--tw-backdrop-opacity) var(--tw-backdrop-saturate) var(--tw-backdrop-sepia);
  }
}
```
and in firefox:

![image](https://user-images.githubusercontent.com/6003803/231226484-e95d4515-ec9d-44fe-97b5-887ceaa923c3.png)

**Reply by wongjn** [reply]
Replied at https://github.com/tailwindlabs/tailwindcss/discussions/10987#discussioncomment-5583190 to keep the reply chain together.

**Comment by nekooee** [comment, marked as answer]
solved my problem with this code:
```
@-moz-document url-prefix() {
    .supports-\[backdrop-filter\]\:tw-backdrop-blur-sm {
        /* styles to override for Firefox */
        backdrop-filter: none !important;
        -webkit-backdrop-filter: none !important;
    }
}
```
